### ISSUE TITLE
Potentially missing Windows language IDs

### ISSUE THREAD

**Issue by Bilal2453** [MicrosoftDocs/typography-issues#1104]
I've noticed while reading the table of possible OpenType's platform specific language IDs assigned on Windows that potentially many macrolanguages are missing.  For instance there are 25 countries (that I know of) who has Arabic as an official language, but only 16 of which are listed, no `Arabic Sudan`/`Arabic Palestine`/`Arabic Djibouti` etc.

Is this list really a complete version? If not either the complete list must be linked and a very obvious warning be added, otherwise an explanation/warning to clear any potential confusion about a missing macrolanguage. 

As far as I am aware though, those regions are listed on Windows as available locales so I would also expect them to be valid in OpenType fonts. 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 9bc5a9d9-70e1-97b3-6926-c176e1a2401d
* Version Independent ID: 1afb0f5b-d1a7-7d69-d175-a5c9382b74be
* Content: [name - Naming table (OpenType 1.9) - Typography](https://learn.microsoft.com/en-us/typography/opentype/spec/name#windows-language-ids)
* Content Source: [typographydocs/opentype/spec/name.md](https://github.com/MicrosoftDocs/typography/blob/live/typographydocs/opentype/spec/name.md)
* Product: **typography**
* GitHub Login: @PeterCon
* Microsoft Alias: **alib**
```[tasklist]
### Tasks
```

```[tasklist]
### Tasks
```

**Comment by Bilal2453** [comment]
On another note, what source is used for the IDs?  Are they always identical to [Windows LCIDs](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c?redirectedfrom=MSDN)?

**Comment by PeterConstable** [comment]
The list in the 'name' table chapter is, indeed, a snapshot of Windows LCIDs as defined in some Windows version. It makes much more sense to point to Microsoft's protocol documentation, which is kept up to date.

**Comment by PeterCon** [comment]
See the [OT 1.9.1 alpha](https://docs.microsoft.com/en-us/typography/opentype/otspec191alpha/ot191alpha) for draft revisions addressing this issue.

**Comment by dscorbett** [comment]
The LCID reference defines some LCIDs as being reserved but also assigned to specific BCP 47 tags. For example, 0x048E is “zh-yue-HK, reserved”. What does that mean? Should they be used in OpenType or not?

Are temporary LCIDs like 0x0C00 and pseudo LCIDs like 0x0501 valid in OpenType?

**Comment by PeterConstable** [comment]
> For example, 0x048E is “zh-yue-HK, reserved”. What does that mean? Should they be used in OpenType or not?

It's outside the scope of the OT spec to define what that means. They could be used in a font, but it's outside the scope of the OT spec to comment on what behaviour should be expected. Interoperable usage in a protocol (which is what we're talking about) is up to the LCID reference to clarify.

>Are temporary LCIDs like 0x0C00 and pseudo LCIDs like 0x0501 valid in OpenType?

The LCID reference is pretty clear about that (highlights added):

>LCID assignments for Locale Names without LCIDs are temporary and are **not suitable for use across a protocol**, or **for interchange between processes** or machines. These temporary LCID assignments are also **unsuitable for tagging persisted data** as the meaning of the LCID assignment will change over time.

All three highlighted bits are applicable here.

**Comment by Bilal2453** [comment]
Oh I forgot to update this issue after I figured some of the things out.

Seems like the main reason for why those languages weren't included in the table is the fact they are effectively unusable, they were added in release 10, and looking at that release, almost all of the languages added has the value `0x1000`.  I was simply looking at language tags and mapping those to Windows LCIDs to be used instead of the table described in the OT specs when I noticed that a ton of languages would map to the same value of `0x1000` which really confused me until I read the LCID specification:
> Locale names that are valid but not associated with a given LCID MAY be assigned the LCID Language
ID 0x1000, if an LCID is requested by the application. [...] any other name will be assigned the LCID Language ID 0x1000. [^1]

> Client releases of Windows 8 or earlier and server releases of Windows Server
2012 or earlier do not recognize these locale names and do not assign them the LCID Language ID
value of 0x1000 for backward compatibility [^2]

From what I have read and understood from one of the MS docs pages (though I couldn't find this one again), those languages/regions are apparently "niche" and for "compatibility" reasons they were assigned the value that used to indicate "User defined LCID".

[^1]: [MS-LCID Section 2.2.1](https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-LCID/%5bMS-LCID%5d-210407.pdf)
[^2]: [MS-LCID Footnote 14 Section 2.2.1](https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-LCID/%5bMS-LCID%5d-210407.pdf)
